### DISCUSSION TITLE
Do I need a bot to read channel's message?

### DISCUSSION THREAD

**Discussion by hilmanski** [discordjs/discord.js#8140] [Q&A] [answered]
Hi.
I'm pretty new to discord itself.

What I want to build is channel reader. Where I can get all the message from a channel.
Do I need a bot for this?
Any reference (which API to use) for this case?

Thanks!

**Comment by nyapat** [comment, marked as answer]
I think you *could* read messages on behalf of a user but I think that's a mild security concern (~~afaik that'd allow the app to read messages anywhere, I wouldn't auth that~~ only happens with local rpc server access). I'm not sure what exactly a "channel reader" will do because you can just read the channel in the app but it does seem like you might want a bot. 

You can get messages by requesting channels/:id/messages with a limit of 100 [via rest](https://discord.com/developers/docs/resources/channel#get-channel-messages) (i.e. rest.get(api route, opts)), or in d.js by just executing [channel.messages.fetch](https://discord.js.org/#/docs/discord.js/main/class/MessageManager?scrollTo=fetch)

**Reply by hilmanski** [reply]
Thanks for you answers and link references. appreciate it.

**For more context:**
I plan to make a 'product' where 'user' can 'connect' their discord to my app.
Then my app will read their channel (messages) and do something with the message.
'Bot' or 'MyApp' won't do anything on that channel/discord. I only want  to read the messages.

**Reply by nyapat** [reply]
Looks like reading messages won't be feasible anyways, you will be restricted to only channels the app creates. 
![image](https://user-images.githubusercontent.com/73502164/174713823-aa50da55-205e-4203-b7b7-e79b2858b2d9.png)
([d-api-docs ref](https://discord.com/developers/docs/topics/oauth2#shared-resources))

**Reply by advaith1** [reply]
if the user is an admin in the server, you should have them add the bot, then use the endpoint with the bot token
